interfaces_test.go

```diff
@@ -215,6 +216,129 @@ func TestEventWithDebugMetaMarshalJSON(t *testing.T) {
 	}
 }
 
+type withCause struct {
+	msg   string
+	cause error
+}
+
+func (w *withCause) Error() string { return w.msg }
+func (w *withCause) Cause() error  { return w.cause }
+
+type customError struct {
+	message string
+}
+
+func (e *customError) Error() string {
+	return e.message
+}
+
+func pointerToInt(i int) *int {
+	return &i
+}
+
+func TestSetException(t *testing.T) {
+	testCases := map[string]struct {
+		exception     error
+		maxErrorDepth int
+		expected      []Exception
+	}{
+		"Single error without unwrap": {
+			exception:     errors.New("simple error"),
+			maxErrorDepth: 1,
+			expected: []Exception{
+				{
+					Value:       "simple error",
+					Type:        "*errors.errorString",
+					Stacktrace:  &Stacktrace{Frames: []Frame{}},
+					ExceptionID: 0,
+				},
+			},
+		},
+		"Nested errors with Unwrap": {
+			exception:     fmt.Errorf("level 2: %w", fmt.Errorf("level 1: %w", errors.New("base error"))),
+			maxErrorDepth: 3,
+			expected: []Exception{
+				{
+					Value:       "base error",
+					Type:        "*errors.errorString",
+					ExceptionID: 0,
+				},
+				{
+					Value:       "level 1: base error",
+					Type:        "*fmt.wrapError",
+					ExceptionID: 1,
+					ParentID:    pointerToInt(0),
+				},
+				{
+					Value:       "level 2: level 1: base error",
+					Type:        "*fmt.wrapError",
+					Stacktrace:  &Stacktrace{Frames: []Frame{}},
+					ExceptionID: 2,
+					ParentID:    pointerToInt(1),
+				},
+			},
```

Comment:
In this case, we should use an ExceptionGroup, so the Mechanism should include `IsExceptionGroup: true`.